### PR TITLE
builtin: built-in func row_count

### PR DESCRIPTION
**Pull request by mccxj**
for https://github.com/pingcap/tidb/issues/3961

### CODE REVIEW

**Comment by shenli** [comment]
@mccxj Thanks!

File: expression/builtin_info_test.go [hunk 1]
```diff
@@ -151,10 +151,16 @@ func (s *testEvaluatorSuite) TestCollation(c *C) {
...
+	sessionVars := ctx.GetSessionVars()
+	sessionVars.PrevAffectedRows = 10
+
 	fc := funcs[ast.RowCount]
-	f, err := fc.getFunction(s.ctx, s.datumsToConstants(types.MakeDatums()))
-	c.Assert(f, IsNil)
-	c.Assert(err, ErrorMatches, "*FUNCTION ROW_COUNT does not exist")
+	f, err := fc.getFunction(ctx, nil)
+	c.Assert(err, IsNil)
+	d, err := evalBuiltinFunc(f, nil)
```
Comment (zz-jason) [inline expression/builtin_info_test.go]: @mccxj We'd better not to use `evalBuiltinFunc()` in unit test.

Comment (zz-jason) [reply expression/builtin_info_test.go]: You can take a look at `TestArithmeticPlus()` at **expression/builtin_arithmetic_test.go** and rewrite this unit test like that.

Comment (mccxj) [reply expression/builtin_info_test.go]: @zz-jason ok. PTAL.

File: executor/prepared.go [hunk 2]
```diff
@@ -397,6 +397,17 @@ func ResetStmtCtx(ctx context.Context, s ast.StmtNode) {
 		sessVars.PrevLastInsertID = sessVars.LastInsertID
 		sessVars.LastInsertID = 0
 	}
+	if sessVars.StmtCtx != nil {
```
Comment (winkyao) [inline executor/prepared.go]: use a separate func to make the code clean.

Comment (winkyao) [reply executor/prepared.go]: Could you remove these code to built func eval?

**Comment by zz-jason** [comment]
@mccxj any update ?

File: executor/prepared.go (hunk 2 above)
Comment (mccxj) [reply executor/prepared.go]: @winkyao row_count relies on affectRows var and the last execute sql type.
I can not find this way to get rid of variable PrevAffectedRows.

**Comment by zz-jason** [comment]
@winkyao PTAL

**Pull request merged**